creativecloud/scripts/utils.js

```diff
@@ -22,18 +22,15 @@
 export const [setLibs, getLibs] = (() => {
   let libs;
   return [
-    (prodLibs) => {
-      const { hostname } = window.location;
-      if (!hostname.includes('hlx.page')
-        && !hostname.includes('hlx.live')
-        && !hostname.includes('localhost')) {
-        libs = prodLibs;
-        return libs;
-      }
-      const branch = new URLSearchParams(window.location.search).get('milolibs') || 'main';
-      if (branch === 'local') return 'http://localhost:6456/libs';
-      if (branch.indexOf('--') > -1) return `https://${branch}.hlx.page/libs`;
-      return `https://${branch}--milo--adobecom.hlx.live/libs`;
+    (prodLibs, location) => {
+      libs = (() => {
+        const { hostname, search } = location || window.location;
+        if (!(hostname.includes('.hlx.') || hostname.includes('local'))) return prodLibs;
+        const branch = new URLSearchParams(search).get('milolibs') || 'main';
+        if (branch === 'local') return 'http://localhost:6456/libs';
+        return branch.includes('--') ? `https://${branch}.hlx.live/libs` : `https://${branch}--milo--adobecom.hlx.live/libs`;
```

Comment:
Branch will always be branch name. as either it will be main or milo branch name from milolibs query param.
If branch name is local the getLibs will again give undefined as libs is not set

Reply:
yes, as mentioned above this is what has been updated in milo code too. I'll check with Chris to confirm the reason behind returning https://${branch}.hlx.live/libs if the branch name has '--' in it.

Reply:
Actually this url looks wrong. It should be https://${branch}.hlx.live/libs is not accessible and it should be https://${branch}--milo--adobecom.hlx.live/libs I'll check with milo team to confirm and update